src/stores/myspaceContext.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { createContext, useContext, useState } from 'react'
+import { DragEndEvent } from '@dnd-kit/core'
+import { arrayMove } from '@dnd-kit/sortable'
+import { WidgetType as AddWidgetType, WidgetReorderInput } from '../graphql.g'
+import { useAnalytics } from 'stores/analyticsContext'
+import { WIDGET_TYPES, MAXIMUM_COLLECTIONS } from 'constants/index'
+import { MySpace, MySpaceWidget, Collection, Widget } from 'types'
+import { useAddCollectionMutation } from 'operations/portal/mutations/addCollection.g'
+import { useAddWidgetMutation } from 'operations/portal/mutations/addWidget.g'
+import { useEditMySpaceMutation } from 'operations/portal/mutations/editMySpace.g'
+
+export type MySpaceContextType = {
+  mySpace: MySpace
+  disableDragAndDrop: boolean
+  setDisableDragAndDrop: React.Dispatch<React.SetStateAction<boolean>>
+  initializeMySpace: (mySpace: MySpace) => void
+  isCollection: (widget: MySpaceWidget) => boolean
+  isGuardianIdeal: (widget: Widget) => boolean
+  isNewsWidget: (widget: Widget) => boolean
+  isFeaturedShortcuts: (widget: Widget) => boolean
+  canAddCollections: boolean
+  canAddNews: boolean
+  canAddGuardianIdeal: boolean
+  canAddFeaturedShortcuts: boolean
+  addNewsWidget: () => void
+  addGuardianIdeal: () => void
+  addFeaturedShortcuts: () => void
+  addNewCollection: () => void
+  handleOnDragEnd?: (event: DragEndEvent) => void
+}
+
+export const MySpaceContext = createContext<MySpaceContextType>({
+  mySpace: [],
+  disableDragAndDrop: false,
+  setDisableDragAndDrop: /* istanbul ignore next */ () => {
+    return
+  },
+  initializeMySpace: /* istanbul ignore next */ () => {
+    return
+  },
+  isCollection: /* istanbul ignore next */ () => {
+    return true || false
```

Comment:
Question: do these need to return `true || false` for the type system?

Reply:
Yep!